src/coreclr/gc/gcpriv.h

```diff
@@ -4954,15 +4955,15 @@ class gc_heap
 
                         size_t new_budget_per_heap = (size_t)(last_budget_per_heap / last_alloc_time * target_alloc_time);
                         new_budget_per_heap = Align (new_budget_per_heap, get_alignment_constant (TRUE));
+                        size_t saved_new_budget_per_heap = new_budget_per_heap;
 
-                        dprintf (6666, ("adjust last budget %Id to %Id (%.3fmb)",
-                            last_budget_per_heap, new_budget_per_heap, (new_budget_per_heap / 1000.0 / 1000.0)));
+                        new_budget_per_heap = max (new_budget_per_heap, bcs_per_heap);
+                        new_budget_per_heap = min (new_budget_per_heap, budget_old_gen_per_heap);
 
-                        if ((new_budget_per_heap >= bcs_per_heap) && (new_budget_per_heap <= budget_old_gen_per_heap))
-                        {
-                            dprintf (6666, ("setting this as the new budget!"));
-                            return new_budget_per_heap;
-                        }
+                        dprintf (6666, ("adjust last budget %Id to %Id->%Id (%.3fmb)",
+                            last_budget_per_heap, saved_new_budget_per_heap, new_budget_per_heap, (new_budget_per_heap / 1000.0 / 1000.0)));
```

Comment:
a few ideas on consistency
- consider changing the dprintf above for `budget_old_gen_per_heap` to also show the before and after for the min/max calculation
- the min and max are in a different order here than above.  However, the difference might be desired as the reasons for the min/max are different in the two places
- if you move `mb` to `gcpriv.h`, then the `gc.cpp` and `gc.priv` code can both use it

Reply:
It looks like the info is all there.  It was just a suggestion because it seemed like the log might be more readable if that first dprintf in this method and this one both had the same style "... <name of value> to {saved_foo}->{foo} ({foo in MB} mb)"